project-base/src/Controller/Front/CartController.php

```diff
@@ -138,13 +149,35 @@ public function indexAction(Request $request)
         ]);
     }
 
-    public function boxAction()
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     */
+    public function boxAction(Request $request)
+    {
+        $orderPreview = $this->orderPreviewFactory->createForCurrentUser();
+
+        return $this->render('Front/Inline/Cart/cartBox.html.twig', [
+            'cart' => $this->cartFacade->findCartOfCurrentCustomerUser(),
+            'productsPrice' => $orderPreview->getProductsPrice(),
+            'isIntentActive' => $request->query->getBoolean('isIntentActive'),
+            'isCartHoverEnable' => !in_array($this->requestStack->getMasterRequest()->get('_route'), self::PAGES_WITH_DISABLED_CART_HOVER, true),
```

Comment:
`getMasterRequest` returns  `Request|null`. I'm not really sure when null can be here, but this code could be extracted into a separate private method as it seems to me. What do you think?

Reply:
Yes, you are right